src/fsharp/utils/FileSystem.fs

```diff
@@ -353,7 +353,6 @@ module MemoryMappedFileExtensions =
             let length = int64 bytes.Length
             trymmf length
                 (fun stream ->
-                    stream.SetLength(stream.Length + length)
                     let span = Span<byte>(stream.PositionPointer |> NativePtr.toVoidPtr, int length)
                     bytes.Span.CopyTo(span)
                     stream.Position <- stream.Position + length
```

Comment:
Do we still need to set the position here too?

Reply:
Yea as `TryFromByteMemory` versions does not reset the stream's position once it's copied.